lib/spack/spack/cmd/license.py

```diff
@@ -89,50 +90,101 @@ def list_files(args):
         print(os.path.join(spack.paths.spack_root, relpath))
 
 
+class LicenseError(object):
+    OLD_LICENSE = 1
```

Comment:
I think you should use an enum object for this, instead of doing it manually

Reply:
I'm not aware of an official python enum (without installing another library). `grep` doesn't immediately reveal any Spack-provided enum class. The way I did it here was (I thought) the idiomatic way to implement an enum in "vanilla" Python 

See also: https://stackoverflow.com/questions/36932/how-can-i-represent-an-enum-in-python

Reply:
The documentation on `enum` says it's new in python 3.4, but I'm also running it right now in python 2.7. So I'm not sure what's going on there.

This link from the python 2.7.6 documentation has the feature, but also says new since 3.4. https://cpython-test-docs.readthedocs.io/en/latest/library/enum.html

Reply:
They're new in 2.7 (not in 2.6) so we can't use this (yet).  I guess I would say let's hold off til enums are possible.

Since these are in `cmd/license.py`, you could reduce the clutter a bit by moving them to module scope.  I think the file is contained enough w/o needing these to be nested in `LicenseError`.

WWIW, what I did in the `SpecParser` for tokens was something like this:

```python
OLD_LICENSE, SPDX_MISMATCH, GENERAL_MISMATCH = range(3)
```

Reply:
https://cpython-test-docs.readthedocs.io/en/latest/library/enum.html indicates to me they are new in `3.4`. My own `2.7.10` install cannot `import enum`. I have moved the error definitions to module scope (although I assigned with `range(1,...)` because I want them all to be `True`).